mrchecker-framework-modules/mrchecker-core-module/src/main/java/com/capgemini/mrchecker/test/core/utils/encryption/controller/DataEncryptionControllerImp.java

```diff
@@ -0,0 +1,78 @@
+package com.capgemini.mrchecker.test.core.utils.encryption.controller;
+
+import static javax.swing.JOptionPane.showMessageDialog;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import javax.swing.*;
+
+import org.jasypt.exceptions.EncryptionOperationNotPossibleException;
+
+import com.capgemini.mrchecker.test.core.base.encryption.providers.DataEncryptionService;
+import com.capgemini.mrchecker.test.core.exceptions.BFSecureModuleException;
+import com.capgemini.mrchecker.test.core.utils.encryption.CryptParams;
+import com.capgemini.mrchecker.test.core.utils.encryption.view.IView;
+
+public class DataEncryptionControllerImp implements DataEncryptionController {
+	private final IView view;
+	
+	public DataEncryptionControllerImp(IView view) {
+		this.view = view;
+	}
+	
+	@Override
+	public void onEncrypt(CryptParams cryptParams) {
+		try {
+			DataEncryptionService.init(new ByteArrayInputStream(cryptParams.getSecret()
+					.getBytes()));
+			view.setEncryptionFieldValue(DataEncryptionService.getInstance()
+					.encrypt(cryptParams.getText()));
+		} catch (BFSecureModuleException e) {
+			showMessageDialog(null, "Could not encrypt because:\n" + e.getMessage());
+			view.setEncryptionFieldValue("");
+		} finally {
+			DataEncryptionService.delInstance();
+		}
+	}
+	
+	@Override
+	public void onDecrypt(CryptParams cryptParams) {
```

Comment:
onEncrypt and onDecrypt have duplicates. Please refactor. You can use functional interface to implement that.